### ISSUE TITLE
[Bug]: Issues with the "Lane" class

### ISSUE THREAD

**Issue by LBlaive** [DATEX-II-EU/DatexII#511]
### What happened?

Comment reported during the Formal Vote ballot in 2018 by FR (no FR074, FR075).
The current definition of the "Lane" class raises two issues:

- The change introduced for Version 3 and consisting in having two different attributes about lane number and lane usage does not actually improve the carriageway and lane model part. Indeed, the corresponding "LaneEnum" enumeration is not only a list of usage but also of physical elements (all lanes, verge, central reservation...). Can a lane no 1 be a "slow vehicle lane" or not? Or vice versa can "a slow vehicle lane" carry a number or not?
- The "laneNumber" attribute is typed as "Integer" which allows having negative numbers or null, which does not correspond to the usage of numbering lanes. Generally speaking, this number starts with 1
Solution: Replace the data type with a specific data type derived from "NonNegativeInteger" with a specific constraint xs:minValue equal to 1.

### Version

3

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by iancornwellmottmac** [comment]
- Lane does allow inconsistent values to be specified. But have we a solution that avoids that?
- I agree with changing the type. If we intend 1-n, then rather than the proposed resolution I would create a PositiveInteger datatype, mapping to xs:positiveInteger, because that is a built-in type in XML Schema that is quite likely to be reused. However, I guess some publishers might wish to say 0 meaning the hard shoulder, in which case we should just use NonNegativeInteger.

**Comment by LBlaive** [comment]
### Issue FR075:
The second point raised in this issue is similar to Issue no [275](https://github.com/DATEX-II-EU/DatexII/issues/275). If the attribute separation is confirmed, it is necessary to refine the corresponding definitions and to clean up the “LaneEnum” enumeration (which could become “LaneUsageEnum”). They need to be considered. As well, the data type of “laneNumber” is to change as proposed in Issue no [275 ](https://github.com/DATEX-II-EU/DatexII/issues/275).

Note 1: Unfortunately, the “laneNumber” definition is the opposite of the one contained in the ETSI Common Date Dictionay (ETSI TS 102 894-2 v2.1.1) (CDD), in its last version.

Note 2: Using”0” for numbering hard shoulder creates an alternative for this geometric element, which might be undesired according to a principle for DATEX II. In ETSI CDD it is used in the “LanePosition” data element but there are two other data elements “DrivingLaneStatus” and “HardShoulderStatus” that are exclusive each from the other.

**Comment by LBlaive** [comment]
### Issue FR074:
The "laneUsage" attribute and the "laneType" enumeration does not fit together. Indeed this enumeration combines physical elements (like e.g. "verge" or "hard shoulder") and traffic management elements like the case of lanes reserved to given vehicle categories (like "slow vehicle lane" or "bus lane"). Moreover, new literals are added through extensions to better accommodate new needs. 
Therefore it is proposed to transform this enumeration into a code list that would be managed separately as a dictionary.

**Comment by iancornwell1** [comment]
A proposal to introduce a codelist, if it is not an existing external one, will also need a more detailed proposal for how to host and manage that codelist. In general I currently think that the primary use of D2Codelist should be for existing externally maintained lists, because for sets that we maintain we did not clearly establish enough benefit of codelists compared to evolving extensions of enumerations.

**Comment by LBlaive** [comment]
The issue of managing code lists is a true issue. It is to note such a mechanism has been implemented for the TN-ITS specification. However, TN-ITS has not defined the concepts of "carriageway" or "lane" transferring them to the location referencing in use. An quick assessment of existing standards shows some basic building blocks like in GDF 5.1 (EN ISO 20524-1) or ISO 17572-4. The most comprehensive for cross-sectional locating is EN ISO 19148 but it does not include a true enumeration or code list. However see C.5.5 of this latter.

**Comment by ingeniumfaber** [comment]
> * Lane does allow inconsistent values to be specified. But have we a solution that avoids that?
> * I agree with changing the type. If we intend 1-n, then rather than the proposed resolution I would create a PositiveInteger datatype, mapping to xs:positiveInteger, because that is a built-in type in XML Schema that is quite likely to be reused. However, I guess some publishers might wish to say 0 meaning the hard shoulder, in which case we should just use NonNegativeInteger.

i agree with the proposal for NonNegativeInteger to keep the hardshoulder management for lane number 0
we also are in favor of keeping the driving lane concept express from the number 1.. n and the lane usage which can be specified by the laneUsage attribute which can be set and managed accordingly to road lane usage.

**Comment by LBlaive** [comment]
Issue FR075: as proposed during the meeting of 20225-01-13 I can propose the following definition of "laneNumber" to extend the existing one for single carriageway roads:
"_The number of the lane, where 1 is nearest the hard shoulder/verge and the number increases towards the central reservation for separate carriageway roads. For single carriageway roads and considering the general orientation of the road element, 1 is nearest the hard shoulder/verge of the side corresponding to the traffic direction aligned with the road orientation and the number increases towards the opposite hard shoulder/verge._"

Issue FR074: some cleanup of the "LaneEnum" enumeration is necessary:

- to separate physical characteristics and usage through two separate enumerations "LaneCharacteristicsEnum" and "LaneUsageEnum" and creation of a second attribute of the "Lane" class ("laneCharacteristics" for physical characteristics);
- cleanup of the existing literals to avoid doublets.

![SupplementaryLocationInformation1](https://github.com/user-attachments/assets/3d00c61d-f5f4-4809-ae73-56d2e3a886fc)

**Comment by iancornwell1** [comment]
On the proposed new definition - it is far over the 255 character limit that we still have for EA tagged values. It could not easily be reduced to 255, so I suggest that we have a much reduced definition, and use the distinctions and clarifications proposed in the new definition in an EA "note" (but not a visible UML comment, it's just metadata) for the element. In the new clarification, even although the last meeting agreed its concepts, I am now not certain of the intent for a single carriageway road, where we use the concept of "orientation". I assumed the intention is for a single number scheme no matter the direction of travel associated with whatever we are describing - no switching of viewpoints, lane 1 is always lane 1 for all contexts. But then I think some people won't understand how to use something called "orientation" to know which (symmetrical) direction is which. When I've worked with map data I could not be any more specific than saying the direction of digitisation of the data, which sometimes seems arbitrary rather than corresponding to any physical or functional property of the physical road. I wouldn't have called that "orientation" without further explanation, so some people may even think that referring to orientation is inviting you to use your viewpoint - in a road that is east-west but you could equally say west-east, in one context you are thinking about eastwards travel east so you choose lane 1 accordingly, in another context somebody is thinking about west so they choose the opposite lane 1. I don't think that is the intention but we should disambiguate. The map data will have arbitrarily chosen one direction in its link-node data, and our lane numbering would relate to that.

On the enums, that looks generally fine to me, except isn't taxiRank a usage (which I assume is something that could be easily changed without physical change)?
